src/onnx/parse_softmaxcrossentropyloss.cpp

```diff
@@ -0,0 +1,307 @@
+    where K is the number of dimensions. Labels element value shall be in range of [0, C).
+    If ignore_index is specified, it may have a value outside [0, C) and the label values should
+    either be in the range [0, C) or have the value ignore_index.
+
+weights (optional, non-differentiable) : T
+    A manual rescaling weight given to each class. If given,
+    it has to be a 1D Tensor assigning weight to each of the classes.
+    Otherwise, it is treated as if having all ones.
+
+Outputs (1 - 2)
+==================
+
+output (differentiable) : T
+    Weighted loss float Tensor. If reduction is 'none', this has the shape of [batch_size],
+    or [batch_size, D1, D2, ..., Dk] in case of K-dimensional loss. Otherwise, it is a scalar.
+
+log_prob (optional, differentiable) : T
+    Log probability tensor. If the output of softmax is prob, its value is log(prob).
+
+Type Constraints
+===================
+    T : tensor(float16), tensor(float), tensor(double), tensor(bfloat16) (Currently not supported in
+MIGX) Constrain input and output types to float tensors.
+
+Tind : tensor(int32), tensor(int64)
+    Constrain target to integer types
+*/
+
+#include <migraphx/onnx/op_parser.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/onnx/checks.hpp>
+#include <migraphx/instruction.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+namespace onnx {
+
+struct parse_softmaxcrossentropyloss : op_parser<parse_softmaxcrossentropyloss>
+{
+    std::vector<op_desc> operators() const { return {{"SoftmaxCrossEntropyLoss"}}; }
+
+    std::vector<instruction_ref> parse(const op_desc& /*opd */,
+                                       const onnx_parser& parser,
+                                       const onnx_parser::node_info& info,
+                                       const std::vector<instruction_ref>& args) const
+    {
+        // Get and handle attributes
+        std::string reduction = "mean";
+        if(contains(info.attributes, "reduction"))
+        {
+            reduction = info.attributes.at("reduction").s();
+            if(not contains({"mean", "sum", "none"}, reduction))
+            {
+                MIGRAPHX_THROW("Invalid reduction mode: " + reduction +
+                               "\n Valid options are [none, mean, sum]");
+            }
+        }
+
+        // ignore_index is optional attribute, assign this as a scalar literal input to the op
+        auto ignore_index =
+            info.add_literal(migraphx::literal(migraphx::shape(shape::int64_type, {1}, {0}), {-1}));
+        bool has_ignore_index = contains(info.attributes, "ignore_index");
+        if(has_ignore_index)
+        {
+            auto ignore_index_val =
+                parser.parse_value(info.attributes.at("ignore_index")).at<int64_t>();
+            ignore_index = info.add_literal(migraphx::literal(
+                migraphx::shape(migraphx::shape::int64_type, {1}, {0}), {ignore_index_val}));
+        }
```

Comment:
Ignore index can be negative and also out of bound. We should check for negative and convert it to positive value. 

If it is out of bound then make `has_ignore_index = false`.

Reply:
This part is done. Sectioned this off as a function called normalize_input_index

Reply:
Done

Reply:
If `ignore_index` is out of bound, that doesn't mean `has_ignore_index = false`. One can have such index present `labels`. As specified in the spec:
> l[i][d1][d2]...[dk]  = 0, when labels[n][d1][d2]...[dk] = ignore_index